### ISSUE TITLE
I't impossible to deprecate a named constant via `define`

### ISSUE THREAD

**Issue by yevgen-2m** [vimeo/psalm#11161]
This is exaple of code:
```
<?php

$someValue = 'abc';

/**
 * @deprecated
 */
define('SOME_VALUE', $someValue);

echo SOME_VALUE;
```

Expected behaviour: Should be warning: defined constant SOME_VALUE is deprecated
Actual  behaviour: No issues

**Comment by yevgen-2m** [comment]
https://psalm.dev/r/65428f72ca
